Negative floating point numbers are interpreted as int

In the REPL, negative floating point numbers seem to be interpreted as int. In this case, math.Abs expects a float64, so the program panics:

```
> import "math"
0xc000162180
> math.Abs(-5.5)  
5.5
> math.Abs(float64(5))     
5
> math.Abs(5.0)
5
> math.Abs(-5.0)
```

Expected result:
```
5
```

Got:
```

1:28: panic
panic: reflect: Call using int as type float64 [recovered]
	panic: reflect: Call using int as type float64

goroutine 11 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc000226000, 0xc000272000)
	/Users/ashneverdawn/go/src/github.com/containous/yaegi/interp/run.go:112 +0x227
panic(0x189d9e0, 0xc000162760)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
reflect.Value.call(0x18b3c60, 0x19fb6f8, 0x13, 0x19c2b15, 0x4, 0xc00000f440, 0x1, 0x1, 0x19c01c0, 0xc0000e9d01, ...)
	/usr/local/go/src/reflect/value.go:377 +0x1457
reflect.Value.Call(0x18b3c60, 0x19fb6f8, 0x13, 0xc00000f440, 0x1, 0x1, 0xc0000e9d88, 0x124e6b8, 0x189c7e0)
	/usr/local/go/src/reflect/value.go:308 +0xa4
github.com/containous/yaegi/interp.callBin.func6(0xc000226000, 0xc00007bd40)
	/Users/ashneverdawn/go/src/github.com/containous/yaegi/interp/run.go:925 +0x1a5
github.com/containous/yaegi/interp.runCfg(0xc000272000, 0xc000226000)
	/Users/ashneverdawn/go/src/github.com/containous/yaegi/interp/run.go:118 +0x70
github.com/containous/yaegi/interp.(*Interpreter).run(0xc00020edc0, 0xc0000f9d00, 0x0)
	/Users/ashneverdawn/go/src/github.com/containous/yaegi/interp/run.go:96 +0x234
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00020edc0, 0xc0000b80f0, 0xf, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/ashneverdawn/go/src/github.com/containous/yaegi/interp/interp.go:364 +0x3f1
github.com/containous/yaegi/interp.(*Interpreter).EvalWithContext.func1(0xc0000f44e0, 0xc00020edc0, 0xc0000b80f0, 0xf, 0xc00000f200, 0xc0001624f0)
	/Users/ashneverdawn/go/src/github.com/containous/yaegi/interp/interp.go:396 +0x6d
created by github.com/containous/yaegi/interp.(*Interpreter).EvalWithContext
	/Users/ashneverdawn/go/src/github.com/containous/yaegi/interp/interp.go:394 +0x14d

```

--- a/_test/bin2.go
+++ b/_test/bin2.go
@@ -1,0 +1,13 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func main() {
+	fmt.Println(math.Abs(-5))
+}
+
+// Output:
+// 5

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1410,7 +1410,35 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 				}
 			}
 			// TODO: Optimisation: avoid allocation if boolean branch op (i.e. '!' in an 'if' expr)
-			n.findex = sc.add(n.typ)
+			if n.child[0].rval.IsValid() && constOp[n.action] != nil {
+				if n.typ == nil {
+					if n.typ, err = nodeType(interp, sc, n); err != nil {
+						return
+					}
+				}
+				n.typ.TypeOf() // init reflect type
+				constOp[n.action](n)
+			}
+			switch {
+			case n.rval.IsValid():
+				n.gen = nop
+				n.findex = -1
+			case n.anc.kind == assignStmt && n.anc.action == aAssign:
+				dest := n.anc.child[childPos(n)-n.anc.nright]
+				n.typ = dest.typ
+				n.findex = dest.findex
+			case n.anc.kind == returnStmt:
+				pos := childPos(n)
+				n.typ = sc.def.typ.ret[pos]
+				n.findex = pos
+			default:
+				if n.typ == nil {
+					if n.typ, err = nodeType(interp, sc, n); err != nil {
+						return
+					}
+				}
+				n.findex = sc.add(n.typ)
+			}
 
 		case valueSpec:
 			n.gen = reset

--- a/interp/run.go
+++ b/interp/run.go
@@ -663,7 +663,7 @@ func call(n *node) {
 				values = append(values, func(f *frame) reflect.Value { return f.data[ind] })
 			}
 		default:
-			if c.kind == basicLit {
+			if c.kind == basicLit || c.rval.IsValid() {
 				var argType reflect.Type
 				if variadic >= 0 && i >= variadic {
 					argType = n.child[0].typ.arg[variadic].val.TypeOf()
@@ -848,7 +848,7 @@ func callBin(n *node) {
 				values = append(values, func(f *frame) reflect.Value { return f.data[ind] })
 			}
 		default:
-			if c.kind == basicLit {
+			if c.kind == basicLit || c.rval.IsValid() {
 				// Convert literal value (untyped) to function argument type (if not an interface{})
 				var argType reflect.Type
 				if variadic >= 0 && i >= variadic {
@@ -2258,13 +2258,16 @@ func recv2(n *node) {
 }
 
 func convertLiteralValue(n *node, t reflect.Type) {
-	if n.kind != basicLit || t == nil || t.Kind() == reflect.Interface {
+	// Skip non-constant values, undefined target type or interface target type.
+	if !(n.kind == basicLit || n.rval.IsValid()) || t == nil || t.Kind() == reflect.Interface {
 		return
 	}
 	if n.rval.IsValid() {
+		// Convert constant value to target type.
 		n.rval = n.rval.Convert(t)
 	} else {
-		n.rval = reflect.New(t).Elem() // convert to type nil value
+		// Create a zero value of target type.
+		n.rval = reflect.New(t).Elem()
 	}
 }
 

